### PR TITLE
feat: Add analytics metadata to Select component

### PR DESCRIPTION
**Pull request by fralongo**
### Description

Similar to https://github.com/cloudscape-design/components/commit/dde24620bceb97c2fd3e59aa45b0e95af8402313

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/internal/components/button-trigger/index.tsx [hunk 1]
```diff
@@ -103,8 +108,16 @@ const ButtonTrigger = (
...
+    action: 'expand',
+    detail: {
+      label: { root: 'self' },
+      expanded: `${!pressed}`,
+    },
+  };
+
   return (
-    <button ref={ref} {...attributes}>
+    <button ref={ref} {...attributes} {...(disabled ? {} : getAnalyticsMetadataAttribute(analyticsMetadata))}>
```
Comment (pan-kot) [inline src/internal/components/button-trigger/index.tsx:120]: Shouldn't it include readonly? When Select is readonly the the dropdown cannot be opened.

File: src/select/__tests__/analytics-metadata.test.tsx [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    expect(getGeneratedAnalyticsMetadata(trigger)).toEqual({
+      action: 'expand',
+      detail: {
+        label: 'select with metadata',
+        expanded: 'false',
+      },
+      ...getMetadataContexts(),
+    });
+  });
+  test('when disabled', () => {
```
Comment (pan-kot) [inline src/select/__tests__/analytics-metadata.test.tsx:113]: If including readonly then should add a test for it too

File: src/internal/components/button-trigger/index.tsx (hunk 1 above)
Comment (fralongo) [reply src/internal/components/button-trigger/index.tsx]: Nicely spotted, addressed!

File: src/select/__tests__/analytics-metadata.test.tsx (hunk 2 above)
Comment (fralongo) [reply src/select/__tests__/analytics-metadata.test.tsx:113]: Done

**Review by pan-kot — APPROVED** [review]

**Pull request merged**